libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1875,6 +1875,38 @@ bool AP_GPS::is_healthy(uint8_t instance) const
            drivers[instance]->is_healthy();
 }
 
+bool AP_GPS::all_heathy() const
+{
+    if (!is_healthy(primary_instance)) {
+        // primary must be healthy
+        return false;
+    }
+
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        // check none primary instances, must be healthy if enabled
+        if ((i != primary_instance) && (get_type(i) != GPS_TYPE_NONE) && !is_healthy(i)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+AP_GPS::GPS_Status AP_GPS::min_status() const
+{
+    // primary
+    AP_GPS::GPS_Status ret = status();
+
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        // check none primary instances
+        if ((i != primary_instance) && (get_type(i) != GPS_TYPE_NONE)) {
```

Comment:
```suggestion
        if (get_type(i) != GPS_TYPE_NONE) {
```

Save some bytes, burn the CPU.  It's not a fast-path call.

And we don't have to try to fix the "English" ;-)